internal/pkg/release/version.go

```diff
@@ -53,7 +54,13 @@ func BuildTime() time.Time {
 
 // Version returns the version of the application.
 func Version() string {
-	return version.GetDefaultVersion()
+	pkgVersion, err := version.GetAgentPackageVersion()
+	if err == nil && pkgVersion != "" {
```

Comment:
I think we should return the error here. When using the `elastic-agent version` command it seems reasonable to return the error, but when reporting the version in a Fleet checkin we can use the unknown suffix since there is no way to return an error there (but we should still log it). We could also just log the error here as well, as long as the error isn't completely lost.

The ECSMetadata function used to obtain the version for checkins looks like it takes a logger as an argument already:

https://github.com/elastic/elastic-agent/blob/b7abc4e2e045d196831cef4a6816828fbb5467f1/internal/pkg/agent/application/info/agent_metadata.go#L140-L142

https://github.com/elastic/elastic-agent/blob/b7abc4e2e045d196831cef4a6816828fbb5467f1/pkg/control/v2/server/server.go#L121-L128

Right now if this function returns an error sometime in the future we'll have no idea why.